src/js/components/AppListComponent.jsx

```diff
@@ -314,22 +366,21 @@ var AppListComponent = React.createClass({
       appsHealthCount: getInitialFilterCounts(HealthStatus)
     };
 
-    // Global search view - only display filtered apps
+    // Global search view - only display filtered apps & groups
     if (this.hasFilters()) {
       appListViewType = AppListViewTypes.APP_LIST;
 
-      nodesSequence = this.filterNodes(lazy(state.apps), filterCounts)
+      nodesSequence = lazy(this.filterNodes(state.apps, filterCounts))
```

Comment:
Yes, I see the problem with the lack of time. :)

Let me quickly explain why your given reduce-example didn't worked out.
As you can see there is this `nodesSequence`. A Sequence is lazy, that said it is only evaluated if you want to access the result of the sequence by converting it to a value (array, whatever..).

.reduce() in the sequence is doing exactly this, it does evaluate the sequence, because it returns a new object, the memo.
With .reduce() you'll receive a real value, not a sequence.

So this would work technically, if you re-lazy the value, but in performance terms I think it is bad:
```JavaScript
nodesSequence = lazy(nodesSequence.reduce((memo, app) => {
    // Push filtered items & new groups in memo
    memo.push( app );
    return memo;
}, []));
```
Not saying this is an ideal approach..